src/masternodes/proposals.h

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) DeFi Blockchain Developers
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef DEFI_MASTERNODES_PROPOSALS_H
+#define DEFI_MASTERNODES_PROPOSALS_H
+
+#include <flushablestorage.h>
+#include <masternodes/res.h>
+#include <script/script.h>
+#include <amount.h>
+#include <serialize.h>
+#include <uint256.h>
+#include <type_traits>
+
+using CPropId = uint256;
+
+enum class CPropType : uint8_t {
+    CommunityFundRequest    = 0x01,
+    BlockRewardRellocation  = 0x02,
+    VoteOfConfidance        = 0x03,
+};
+
+enum class CPropStatusType : uint8_t {
+    Voting    = 0x01,
+    Rejected  = 0x02,
+    Completed = 0x03,
+};
+
+enum class CPropVoteType : uint8_t {
+    VoteYes     = 0x01,
+    VoteNo      = 0x02,
+    VoteNeutral = 0x03,
+};
+
+std::string CPropTypeToString(const CPropType status);
+std::string CPropVoteToString(const CPropVoteType status);
+std::string CPropStatusToString(const CPropStatusType status);
+
+struct CCreatePropMessage {
+    CPropType type;
+    CScript address;
+    CAmount nAmount;
+    uint8_t nCycles;
+    std::string title;
```

Comment:
Is there a limit on number of chars? I would propose a limit here, perhaps 256 chars.

Reply:
Limit is 128 chars